api/blog.js

```diff
@@ -42,26 +49,46 @@ function fetchBlogPostPageTemplate(locale) {
 }
 
 async function fetchPreviousBlogPostSlug(locale, date) {
-  const entries = await fetchEntriesForContentType('blogPost', {
+  let entries = await fetchEntriesForContentType('blogPost', {
     locale,
     limit: 1,
     order: '-fields.date',
     'fields.date[lt]': date,
     'fields.slug[exists]': true,
   })
 
+  if (entries.length === 0) {
+    entries = await fetchEntriesForContentType('blogPost', {
+      locale: 'de',
+      limit: 1,
+      order: '-fields.date',
+      'fields.date[lt]': date,
+      'fields.slug[exists]': true,
+    })
+  }
+
   return (entries && entries.length > 0) ? entries[0].fields.slug : null
 }
 
 async function fetchNextBlogPostSlug(locale, date) {
-  const entries = await fetchEntriesForContentType('blogPost', {
+  let entries = await fetchEntriesForContentType('blogPost', {
     locale,
     limit: 1,
     order: 'fields.date',
     'fields.date[gt]': date,
     'fields.slug[exists]': true,
   })
```

Comment:
How about moving the common options to a separate variable, to make it more visible what the difference between the two calls is?

```javascript
const fetchParams = {
  limit: 1,
  order: 'fields.date',
  'fields.date[gt]': date,
  'fields.slug[exists]': true,
}

let entries = await fetchEntriesForContentType('blogPost', { locale, ...fetchParams })

if (entries.length === 0) {
  entries = await fetchEntriesForContentType('blogPost', { locale: 'de', ...fetchParams })
}
```

Reply:
I knew you were going to comment on that. Pure laziness on my part. ;)